resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/CircuitBreakerStateMachine.java

```diff
@@ -293,6 +291,9 @@ private void stateTransition(State newState,
         UnaryOperator<CircuitBreakerState> newStateGenerator) {
         CircuitBreakerState previousState = stateReference.getAndUpdate(currentState -> {
             StateTransition.transitionBetween(getName(), currentState.getState(), newState);
+            if (currentState instanceof OpenState) {
```

Comment:
I'm not happy that this state-specific code is in generic method of the CircuitBreakerStateMachine.
Maybe we can add a `postTransitionHook` or `preTransitionHook` to states which are invoked here in a generic form?

Reply:
Yes right. that would be better. I'll add `preTransitionHook`.